File: core/src/galenscovell/hinterstar/processing/Node.kt
package galenscovell.hinterstar.processing

import galenscovell.hinterstar.generation.interior.Tile


class Node(private val tile: Tile) {
    private var parent: Node? = null
    private var costFromStart: Double = 0.0
    private var totalCost: Double = 0.0



    /********************
     *     Getters     *
     ********************/
    fun getCostFromStart(): Double {
        return costFromStart
    }

    fun getTotalCost(): Double {
        return totalCost
    }

    fun getTile(): Tile {
        return tile
    }

    fun getParent(): Node? {
        return parent
    }



    /********************
     *     Setters     *
     ********************/
    fun setCostFromStart(cost: Double): Unit {
        costFromStart = cost
    }

    fun setTotalCost(cost: Double): Unit {
        totalCost = cost
    }

    fun setParent(node: Node): Unit {
        parent = node
    }
}

File: core/src/galenscovell/hinterstar/processing/Node.scala
package galenscovell.hinterstar.processing

import galenscovell.hinterstar.generation.interior.Tile


class Node(tile: Tile) {
  private var parent: Node = _
  private var costFromStart, totalCost: Double = _



  /********************
    *     Getters     *
    ********************/
  def getCostFromStart: Double = {
    costFromStart
  }

  def getTotalCost: Double = {
    totalCost
  }

  def getTile: Tile = {
    tile
  }

  def getParent: Node = {
    parent
  }



  /********************
    *     Setters     *
    ********************/
  def setCostFromStart(cost: Double): Unit = {
    costFromStart = cost
  }

  def setTotalCost(cost: Double): Unit = {
    totalCost = cost
  }

  def setParent(node: Node): Unit = {
    parent = node
  }
}
